bundles/umljava/tools.vitruv.applications.umljava.uml2java/src/tools/vitruv/applications/umljava/uml2java/UmlToJavaClassifier.reactions

```diff
@@ -276,7 +310,10 @@ reaction UmlDataTypeCreated {
     after element uml::DataType created and inserted in uml::Package[packagedElement] 
     with !(newValue instanceof PrimitiveType)
     	&& !(newValue instanceof Enumeration) 
-    call createJavaClass(newValue)
+    call {
+        createOrFindJavaClass(newValue)
```

Comment:
Since the sequence `createOrFindJavaClass` and `createRootJavaClass` now has to be called at several places and in fact from a logical point of view `createRootJavaClass` should be covered by `createOrFindJavaClass`, because it is also a kind of Java class creation for a file that was not found, I would propose to combine both routines in the `createOrFindJavaClass` routine. More precisely, I would keep the first two match statements in that routine and than call two subroutines, one for `createOrFindJavaClassInPackage` or something comparable, and one for the exsting `createRootJavaClass`. Then the logic should be clearer and it would be only necessary to call one routine.